File: php/test/MineSweeperTest.php
<?php

require_once dirname(__FILE__).'/../src/MineSweeper.php';

define(VALID_INPUT, <<<EOD
4 4
*...
....
.*..
....
3 5
**...
.....
.*...
0 0
EOD
);

define(EXPECTED_OUTPUT, <<<EOD
Field #1:
*100
2210
1*10
1110

Field #2:
**100
33200
1*100


EOD
);

class MineSweeperTest extends PHPUnit_Framework_TestCase {

	public function test_aceptacion() {
        $mineSweeper = new MineSweeper();
        $this->assertEquals(EXPECTED_OUTPUT, $mineSweeper->process(VALID_INPUT));
    }

}

File: java/test/MineSweeperTest.java
import static org.junit.Assert.*;

import org.junit.Test;

public class MineSweeperTest {

	@Test
	public void test() {
		String aValidInput = "4 4\n*...\n....\n.*..\n....\n3 5\n**...\n.....\n.*...\n0 0\n";
		String theExpectedOutput = "Field #1:\n*100\n2210\n1*10\n1110\n\nField #2:\n**100\n33200\n1*100\n";

		MineSweeper mineSweeper = new MineSweeper();

		assertEquals(theExpectedOutput, mineSweeper.process(aValidInput));
	}

}
